src/createClient.ts

```diff
@@ -501,15 +503,37 @@ export class Client<TDocuments extends PrismicDocument = PrismicDocument> {
 	 */
 	constructor(repositoryNameOrEndpoint: string, options: ClientConfig = {}) {
 		if (isRepositoryEndpoint(repositoryNameOrEndpoint)) {
-			if (
-				process.env.NODE_ENV === "development" &&
-				/\.prismic\.io\/(?!api\/v2\/?)/.test(repositoryNameOrEndpoint)
-			) {
-				throw new PrismicError(
-					"@prismicio/client only supports Prismic Rest API V2. Please provide only the repository name to the first createClient() parameter or use the getRepositoryEndpoint() helper to generate a valid Rest API V2 endpoint URL.",
-					undefined,
-					undefined,
-				);
+			if (process.env.NODE_ENV === "development") {
+				/**
+				 * Matches non-API v2 `.prismic.io` endpoints
+				 * {@link https://regex101.com/r/xRsavu/1}
+				 */
```

Comment:
Style: I typically use `/* ... */` comments only for TSDoc (i.e. ones that should be public) and `//` for internal comments. Following that convention makes it easy to put TSDocs right next to internal comments, like this:

```typescript
/* A public comment. */
// TODO: A private comment
function foo() {
  return "bar"
}
```

`foo()` will retain the TSDoc even with the `//` comment between the documentation and the function.